gymnasium/core.py

```diff
@@ -52,6 +52,8 @@ class Env(Generic[ObsType, ActType]):
     render_mode: str | None = None
     reward_range = (-float("inf"), float("inf"))
     spec: EnvSpec | None = None
+    spec_stack: SpecStack | None = None
```

Comment:
I think this should be a property rather than an attribute. 
If you use the function proposed in the issue then you don't need to include any of the `self.spec_stack.append_wrapper(self)`

Reply:
In order to avoid having to explicitly define whether a wrapper is an outer layer or not, there would need to be a callback in every wrapper like how `gym.utils.EzPickle.__init__(self)` is used. This is why we need `append_wrapper()`. `spec_stack()` (the function proposed in the issue), on the other hand, does not call wrappers at all; it recursively collects metadata from the unwrapping of wrappers.

Reply:
What do you mean by an outer layer? In my head, I would have thought, you receive a sequence of wrapper specs + a env spec and you just generate each of the wrappers in order. Therefore, the `env.spec_stack` and `EzPickle` as part of the function definition is all that is needed.
Could we move the json based code for loading and saving info to minari not gymnasium? 

Personally, I think all gymnasium should do in this PR is add `EzPickle` to wrappers, add a `spec_stack` function and add functional to make for a sequence of wrapper and env spec. I would like minari to be as minimum as possible in gymnasium.

Reply:
I think you have misread me as I wish `spec_stack` to be a function (technically a property) not an attribute. 
For `spec` this is an attribute in `Env` but a function (technically a property) in `Wrapper`.

This is my proposal from the issue, modified for `Env` and `Wrapper`
```python
class Env:
    @property
    def spec_stack(self) -> tuple[Union[EnvSpec, WrapperSpec]]:
         assert self.spec is not None
         return (self.spec,)
     
class Wrapper:
    @property
    def spec_stack(self) -> tuple[Union[EnvSpec, WrapperSpec]]:
          assert hasattr(self, "_ezpickle_kwargs") and hasattr(self, "_ezpickle_args")
          wrapper_spec = WrapperSpec(self.__name__, self.__module__ + "." + self.__name__, self._ezpickle_args, self.ezpickle_kwargs)
          return (wrapper_spec,) + self.env.spec_stack
```

Therefore, this is no need for the `append_wrapper`, just that the wrapper class inherit from `EzPickle` and the environment has a `spec`. It might be possible to wrap the into the `SpecStack` class but this would be for simplicity rather than 2/3 separate functions.

On where the code goes minari or gymnasium. It makes more sense now why we would keep it in gymnasium. I would just have the code in `utils` rather than `registration.py`.

Reply:
Wow, I overcomplicated this.

Ok, I get you now. I've modified this PR accordingly (still need to update the tutorial, so ignore that). Please lmk your thoughts.